pkg/apis/objectbucket.io/v1alpha1/objectbucketclaim_types.go

```diff
@@ -1,13 +1,37 @@
 package v1alpha1
 
 import (
+	"github.com/aws/aws-sdk-go-v2/service/s3"
 	"k8s.io/api/core/v1"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 // ObjectBucketClaimSpec defines the desired state of ObjectBucketClaim
 type ObjectBucketClaimSpec struct {
+	// StorageClass names the StorageClass object representing the desired provisioner and parameters
 	StorageClass string
+	// BucketName (not recommended) the name of the bucket.  Caution!
+	// In-store bucket names may collide across namespaces.  If you define
+	// the name yourself, try to make it as unique as possible.
+	BucketName string `json:"bucketName,omityempty"`
+	// GenerateBucketName (recommended) a prefix for a bucket name to be
+	// followed by a hyphen and 5 random characters. Protects against
+	// in-store name collisions.
+	GeneratBucketName string `json:"generatBucketName,omitempty"`
+	// SSL whether connection to the bucket requires SSL authentication or not
+	SSL bool `json:"ssl"`
+	// AWS S3 predefined bucket ACLs.
+	// Available BucketCannedACLs are:
+	//    BucketCannedACLPrivate
+	//    BucketCannedACLPublicRead
+	//    BucketCannedACLPublicReadWrite
+	//    BucketCannedACLAuthenticatedRead
+	CannedBucketACL s3.BucketCannedACL `json:"cannedBucketAcl"`
+	// Versioned determines if versioning is enabled
+	Versioned bool `json:"versioned"`
+	// AdditionalConfig gives non-AWS S3 providers a location to set
+	// proprietary config values (tenant, namespace, etc)
+	AdditionalConfig map[string]interface{} `json:"additionalConfig"`
```

Comment:
can the Phase string constants be defined in only 1 file?

Reply:
Do you mean using them for both OBC and OB phases?  In the current case, yes as they're identical.  We need to really decide on what they phases should be.